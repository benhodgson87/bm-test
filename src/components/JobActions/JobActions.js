import React from 'react'
import PropTypes from 'prop-types'
import { DropdownButton, MenuItem } from 'react-bootstrap'

const JobActions = ({ id, handleEdit, handleDelete }) => (
  <DropdownButton bsStyle="primary" title="Actions" className="job-actions-handle">
    <MenuItem eventKey="1" onClick={() => handleEdit(id)} className="job-actions-edit">
      Edit
    </MenuItem>
    <MenuItem eventKey="2" onClick={() => handleDelete(id)} className="job-actions-delete">
      Delete
    </MenuItem>
  </DropdownButton>
)

JobActions.propTypes = {
  id: PropTypes.number.isRequired,
  handleEdit: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
}

export default JobActions
