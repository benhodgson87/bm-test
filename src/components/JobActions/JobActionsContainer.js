import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { deleteJob } from '../../store/jobs/thunks'

import JobActions from './JobActions'

export class JobActionsContainer extends Component {
  static propTypes = {
    id: PropTypes.number.isRequired,
    deleteById: PropTypes.func.isRequired,
  }

  handleEdit(id) {
    console.log(`Pop a modal for editing ${id} record data`)
  }

  handleDelete(id) {
    const { deleteById } = this.props
    const isConfirmed = window.confirm('Are you sure you want to delete this record?') // eslint-disable-line no-alert
    if (isConfirmed) deleteById(id)
  }

  render() {
    const { id } = this.props
    return (
      <JobActions
        id={id}
        handleEdit={e => this.handleEdit(e)}
        handleDelete={e => this.handleDelete(e)}
      />
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  deleteById: id => dispatch(deleteJob(id)),
})

export default connect(null, mapDispatchToProps)(JobActionsContainer)
