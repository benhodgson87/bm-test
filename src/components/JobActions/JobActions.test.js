import React from 'react'
import { mount } from 'enzyme'

import JobActions from './JobActions'

const handleEdit = jest.fn()
const handleDelete = jest.fn()

const baseProps = {
  id: 123,
  handleEdit,
  handleDelete,
}

const renderProps = props => Object.assign({}, baseProps, props)

describe('JobActions presentational component', () => {
  test('renders as expected', () => {
    const wrapper = mount(<JobActions {...renderProps({})} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('Fires an edit callback on click', () => {
    const wrapper = mount(<JobActions {...renderProps()} />)
    wrapper
      .find('.job-actions-edit')
      .find('a')
      .simulate('click')
    expect(handleEdit).toHaveBeenCalledWith(123)
  })

  test('Fires a delete callback on click', () => {
    const wrapper = mount(<JobActions {...renderProps()} />)
    wrapper
      .find('.job-actions-delete')
      .find('a')
      .simulate('click')
    expect(handleDelete).toHaveBeenCalledWith(123)
  })
})
