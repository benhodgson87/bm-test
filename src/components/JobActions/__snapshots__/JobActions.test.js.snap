// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobActions presentational component renders as expected 1`] = `
<JobActions
  handleDelete={[MockFunction]}
  handleEdit={[MockFunction]}
  id={123}
>
  <DropdownButton
    bsStyle="primary"
    className="job-actions-handle"
    title="Actions"
  >
    <Uncontrolled(Dropdown)
      bsStyle="primary"
    >
      <Dropdown
        bsClass="dropdown"
        bsStyle="primary"
        componentClass={[Function]}
        onToggle={[Function]}
      >
        <ButtonGroup
          block={false}
          bsClass="btn-group"
          bsStyle="primary"
          className="dropdown"
          justified={false}
          vertical={false}
        >
          <div
            className="dropdown btn-group btn-group-primary"
          >
            <DropdownToggle
              bsClass="dropdown-toggle"
              bsRole="toggle"
              bsStyle="primary"
              className="job-actions-handle"
              key=".0"
              onClick={[Function]}
              onKeyDown={[Function]}
              open={false}
              useAnchor={false}
            >
              <Button
                active={false}
                aria-expanded={false}
                aria-haspopup={true}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                className="job-actions-handle dropdown-toggle"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="job-actions-handle dropdown-toggle btn btn-primary"
                  disabled={false}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  type="button"
                >
                  Actions
                   
                  <span
                    className="caret"
                  />
                </button>
              </Button>
            </DropdownToggle>
            <DropdownMenu
              bsClass="dropdown-menu"
              bsRole="menu"
              key=".1"
              onClose={[Function]}
              onSelect={[Function]}
              pullRight={false}
            >
              <RootCloseWrapper
                disabled={true}
                event="click"
                onRootClose={[Function]}
              >
                <ul
                  className="dropdown-menu"
                  role="menu"
                >
                  <MenuItem
                    bsClass="dropdown"
                    className="job-actions-edit"
                    disabled={false}
                    divider={false}
                    eventKey="1"
                    header={false}
                    key=".0"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onSelect={[Function]}
                  >
                    <li
                      className="job-actions-edit"
                      role="presentation"
                    >
                      <SafeAnchor
                        componentClass="a"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          role="menuitem"
                          tabIndex="-1"
                        >
                          Edit
                        </a>
                      </SafeAnchor>
                    </li>
                  </MenuItem>
                  <MenuItem
                    bsClass="dropdown"
                    className="job-actions-delete"
                    disabled={false}
                    divider={false}
                    eventKey="2"
                    header={false}
                    key=".1"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onSelect={[Function]}
                  >
                    <li
                      className="job-actions-delete"
                      role="presentation"
                    >
                      <SafeAnchor
                        componentClass="a"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="menuitem"
                        tabIndex="-1"
                      >
                        <a
                          href="#"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          role="menuitem"
                          tabIndex="-1"
                        >
                          Delete
                        </a>
                      </SafeAnchor>
                    </li>
                  </MenuItem>
                </ul>
              </RootCloseWrapper>
            </DropdownMenu>
          </div>
        </ButtonGroup>
      </Dropdown>
    </Uncontrolled(Dropdown)>
  </DropdownButton>
</JobActions>
`;
